src/Cassandra/Metadata.cs

```diff
@@ -142,29 +144,49 @@ public ICollection<Host> AllHosts()
             return Hosts.ToCollection();
         }
 
-
         public IEnumerable<IPEndPoint> AllReplicas()
         {
             return Hosts.AllEndPointsToCollection();
         }
 
+        private bool AnotherRebuildCompleted(long currentCounter)
+        {
+            var lastCounter = Interlocked.Read(ref _counterRebuild);
+            if (lastCounter != currentCounter)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool RebuildNecessary(long currentCounter)
+        {
+            if (AnotherRebuildCompleted(currentCounter))
```

Comment:
Suggestion: this can be simplified by using `Interlocked.CompareExchange()`, with the current value as _comparand_ and use `AnotherRebuildCompleted()` only for single keyspace update.

Reply:
I should have done that to begin with. Changed!